### PR TITLE
allow TokenType to be an object without requiring a subclass

### PR DESCRIPTION
**Pull request by roastario**

### CODE REVIEW

File: modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt [hunk 1]
```diff
@@ -14,38 +14,42 @@ import net.corda.core.transactions.LedgerTransaction
...
 }
 
 object RUB : Ruble()
 
 
-open class THING : TokenType {
-    override val tokenIdentifier: String = "PTK"
-    override val fractionDigits: Int = 0
-
+open class PhoBowl : TokenType("PTK", 0) {
```
Comment (kasiastreich) [inline modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: and I know what you had for lunch

File: modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt [hunk 2]
```diff
@@ -14,38 +14,36 @@ import net.corda.core.transactions.LedgerTransaction
...
-        get() = "рубль"
-    override val fractionDigits: Int
-        get() = 0
-
-    override fun toString(): String {
-        return "Ruble(tokenIdentifier: ${tokenIdentifier}, fractionDigits: ${fractionDigits})"
+open class Ruble : TokenType("рубль", 0) {
+
+
+    override fun equals(other: Any?): Boolean {
```
Comment (kasiastreich) [inline modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: why equals override?

File: modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt [hunk 3]
```diff
@@ -14,38 +14,36 @@ import net.corda.core.transactions.LedgerTransaction
...
-object PTK : THING()
+object PTK : PhoBowl()
 
 
-data class Appartment(
-        override val tokenIdentifier: String = "FOO"
-) : TokenType {
-    override val fractionDigits: Int = 0
-}
+data class Appartment(val id: String = "Foo") : TokenType(id, 0)
```
Comment (kasiastreich) [inline modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: oh man, not again!!!

Comment (roger3cev) [reply modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: Haha

Comment (roger3cev) [reply modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: Apartment only has one "p". How did you miss that @kasiastreich ?!

Comment (kasiastreich) [reply modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: I know how delicate Stefano is, I didn't want to discourage him.

Comment (gendal) [reply modules/contracts-for-testing/src/main/kotlin/com/r3/corda/lib/tokens/testing/states/Rubles.kt]: Just be glad he didn't get all clever and write it as condomminnnniuuum

**Pull request merged**